lib/OpenQA/Task/Needle/Delete.pm

```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Task::Needle::Delete;
+use Mojo::Base 'Mojolicious::Plugin';
+
+use OpenQA::Utils;
+use Scalar::Util 'looks_like_number';
+
+sub register {
+    my ($self, $app) = @_;
+    $app->minion->add_task(delete_needles => sub { _delete_needles($app, @_) });
+}
+
+sub _delete_needles {
+    my ($app, $minion_job, $args) = @_;
+
+    # prevent multiple save_needle and delete_needles tasks to run in parallel
+    return $minion_job->fail({error => 'Another save or delete needle job is ongoing. Try again later.'})
+      unless my $guard = $app->minion->guard('limit_needle_task', 300);
```

Comment:
Shouldn't you retry if you just want to try again later?

Reply:
Ah, or is this meant as a message to be relayed to the user via job result?

Reply:
If that's the case i think you should use `$minion_job->finish` though, since failed jobs are considered real failures in Minion and need to be manually reviewed/removed by an admin with the ui.

Reply:
Now that we have monitoring for failed Gru/Minion jobs, this would also be a problem there.

Reply:
Yes, this message is actually presented to the user. It is not the best user-experience but I also don't want to over-complicate things for now.

I'll change it to use finish and also adjust the that for saving needles.

Reply:
Yes, that should work.